examples/python/projdata_visualisation/ProjDataVisualisation.py

```diff
@@ -0,0 +1,265 @@
+            ax.xaxis.set_label_text("Tangential positions")
+        elif self.viewgram_radio_button.isChecked():
+            image = self.get_viewgram_numpy_array()
+            ax.title.set_text(
+                f"Sinogram - Segment: {self.UI_groupbox_projdata_dimensions.value(ProjDataDims.SEGMENT_NUM)},"
+                f"View Number: {self.UI_groupbox_projdata_dimensions.value(ProjDataDims.VIEW_NUMBER)}")
+            ax.yaxis.set_label_text("Axial positions")
+            ax.xaxis.set_label_text("Tangential positions")
+        else:
+            msg = f"Error: No radio button is checked... How did you get here?\n"
+            raise Exception(msg)
+
+        # display the image
+        ax.imshow(image,
+                  # cmap='gray'
+                  )
+        self.display_image_matplotlib_canvas.draw()
+
+    def get_sinogram_numpy_array(self):
+        """
+        This function returns the sinogram numpy array based on the current UI configuration parameters for segment 
+        number and axial position number. 
+        """
+        if self.stir_interface.projdata is None:
+            return None
+
+        axial_pos = self.UI_groupbox_projdata_dimensions.value(ProjDataDims.AXIAL_POS)
+        return self.stir_interface.as_numpy(
+            self.stir_interface.segment_data.get_sinogram(axial_pos))
+
+    def get_viewgram_numpy_array(self):
+        if self.stir_interface.projdata is None:
+            return None
+
+        view_num = self.UI_groupbox_projdata_dimensions.value(ProjDataDims.VIEW_NUMBER)
+        return self.stir_interface.as_numpy(self.stir_interface.segment_data.get_viewgram(view_num))
+
+    def browse_file_system_for_projdata(self):
+        initial = self.projdata_filename_box.text()
+        filename = QFileDialog.getOpenFileName(self, "Open ProjData File", "", "ProjData Files (*.hs)")
+        if len(filename[0]) > 0:
+            self.projdata_filename_box.setText(filename[0])
+            self.load_projdata()
+        else:
+            self.projdata_filename_box.setText(initial)
+
+    def load_projdata(self, filename=None):
+        """
+        This function loads the projdata file and updates the UI.
+        """
+        if filename is not None and filename != "":
+            self.projdata_filename_box.setText(filename)
+
+        self.stir_interface.load_projdata(self.projdata_filename_box.text())
+        self.refresh_UI_configuration()
+
+    def set_projdata(self, projdata):
+        self.stir_interface.set_projdata(projdata)
+        self.refresh_UI_configuration()
+        self.projdata_filename_box.setText("ProjData set externally.")
+
+
+def OpenProjDataVisualisation(filename=None, projdata=None):
+    app = QApplication([])
+    gallery = ProjDataVisualisationWidgetGallery()
+    
+    if isinstance(filename, str):
+        gallery.load_projdata(filename)
+    elif projdata is not None:
+        gallery.set_projdata(projdata)
```

Comment:
that's rather non-intuitive. I'd probably just remove the `filename` argument, or (probably better) do
```suggestion
def OpenProjDataVisualisation(projdata=None):
    app = QApplication([])
    gallery = ProjDataVisualisationWidgetGallery()
    
    if isinstance(projdata, str):
        gallery.load_projdata(projdata)
    elif  isinstance(projdata, str):
        gallery.set_projdata(projdata)
     elif projdata is not None:
       throw runtimeerrorsomehow
```

Reply:
could also use a docstring if you don't mind.